crates/ruff_python_ast/src/helpers.rs

```diff
@@ -1438,32 +1438,18 @@ pub fn typing_optional(elt: Expr, binding: Name) -> Expr {
 
 /// Format the expressions as a `typing.Union`-style union.
 pub fn typing_union(elts: &[Expr], binding: Name) -> Expr {
-    fn tuple(elts: &[Expr], binding: Name) -> Expr {
```

Comment:
I'm not familiar with the code so this is probably a dumb question but the the old implementation applied this function recursively. Is it correct that this is no longer needed?

Reply:
Sorry, it's hard to recall without digging deeper. But, yes, I'm guessing it's something to do with the recursive nature of the `|` operator. Notice that this is very similar to the `pep_604_union` version above.